### PR TITLE
W-13829786: Avoid reading mule-module.properties for modules with module-info

### PR DESCRIPTION
**Pull request by elrodro83**
Fixing odd things in this tests that cause them to fail in a more restrictive modularized environment.

### CODE REVIEW

File: tracing/pom.xml [hunk 1]
```diff
@@ -214,11 +214,6 @@
             <artifactId>mule-tracer-exporter-configuration-api</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <groupId>org.mule.runtime</groupId>
-            <artifactId>mule-tracer-configuration-api</artifactId>
-            <version>${project.version}</version>
-        </dependency>
```
Comment (asanguinetti) [inline tracing/pom.xml:221]: What does this have to do with all the rest of the changes?

Comment (elrodro83) [reply tracing/pom.xml:221]: This was wrongly placed there and was working by chance. Fixing odd things in this tests that cause them to fail in a more restrictive modularized environment.

Comment (asanguinetti) [reply tracing/pom.xml:221]: Sorry I missed the PR description
